File: mywebsite-parent-v2.0/mywebsite-manager-web/src/main/java/xyz/lirui123/mywebsite/manager/service/CommentService.java
package xyz.lirui123.mywebsite.manager.service;


import org.springframework.stereotype.Component;
import xyz.lirui123.mywebsite.pojo.TbComment;
import xyz.lirui123.mywebsite.response.ResponseResult;

public interface CommentService {

    /**
     * 根据id查询评论
     * @param id
     * @return
     */
    public abstract TbComment findOneById(Long id);

    /**
     * 条件分页搜索
     * @param tbComment
     * @param pageNum
     * @param pageSize
     * @return
     */
    public abstract ResponseResult search(TbComment tbComment, int pageNum, int pageSize);


    /**
     * 添加评论
     * @param tbComment
     * @return
     */
    public abstract ResponseResult add(TbComment tbComment);

    /**
     * 批量审核
     * @param ids
     * @param status
     * @return
     */
    public abstract ResponseResult updateStatusIds(Long[] ids, String status);

    /**
     * 批量删除（关联删除）
     * @param ids
     * @return
     */
    public abstract ResponseResult deleteIds(Long[] ids);

    /**
     * 修改评论信息
     * @param tbComment
     * @return
     */
    public abstract ResponseResult update(TbComment tbComment);

    /**
     * 通过博客id查询所有评论
     * @param blogId
     * @return
     */
    public abstract ResponseResult getCommentByBlogId(Long blogId);

    /**
     * 通过动态id查询所有评论
     * @param trendsId
     * @return
     */
    public abstract ResponseResult getCommentByTrendsId(Long trendsId);

    /**
     * 点赞
     * @param id
     * @return
     */
    public abstract ResponseResult updateCommentPraseCount(Long id);
}

File: mywebsite-parent-v2.0/mywebsite-protal-web/src/main/webapp/angularjs/js/service/commentService.js
app.service('commentService', function ($http) {

    /**
     * 条件分页搜索
     * @param searchEntity
     * @param page
     * @param rows
     * @returns {*}
     */
    this.search = function (searchEntity, page, rows) {
        return $http.post('../comment/search?page=' + page + '&rows=' + rows, searchEntity);
    };

    /**
     * 根据id查询评论
     * @param id
     * @returns {*}
     */
    this.findOneById = function (id) {
        return $http.get('../comment/findOneById?id=' + id);
    };

    /**
     * 更新评论
     * @param entity
     * @returns {*}
     */
    this.update = function (entity) {
        return $http.post('../comment/update', entity);
    };

    /**
     * @param entity
     * @returns {*}
     */
    this.add = function (entity) {
        return $http.post('../comment/add', entity);
    };

    /**
     * 批量删除
     * @param ids
     * @returns {*}
     */
    this.deleteIds = function (ids) {
        return $http.get('../comment/deleteIds?ids='+ids);
    };

    /**
     * 批量审核
     * @param ids
     * @param status
     * @returns {*}
     */
    this.updateStatusIds = function (ids, status) {
        return $http.get('../comment/updateStatusIds?ids=' + ids + '&status=' + status);
    };

    /**
     * 根据博客id查询评论信息
     * @param blogId
     * @returns {*}
     */
    this.getCommentByBlogId = function (blogId) {
        return $http.get('../comment/getCommentByBlogId?blogId=' + blogId);
    };

    /**
     * 根据动态id查询评论信息
     * @param trendsId
     * @returns {*}
     */
    this.getCommentByTrendsId = function (trendsId) {
        return $http.get('../comment/getCommentByTrendsId?trendsId=' + trendsId);
    };

    /**
     * 点赞
     * @param id
     * @returns {*}
     */
    this.updateCommentPraseCount = function (id) {
        return $http.get('../comment/updateCommentPraseCount?id=' + id);
    };
});